repology/fetchers/fetchers/repodata.py

```diff
@@ -35,6 +35,15 @@ def __init__(self, url: str, fetch_timeout: int = 60):
 
     def _do_fetch(self, statefile: AtomicFile, persdata: PersistentData, logger: Logger) -> bool:
 
+        # if given url is a mirror.list convert it to a baseurl
+        if self.url.endswith('mirror.list'):
+            baseurls = do_http(self.url, check_status=True, timeout=self.fetch_timeout).text
+            baseurl = baseurls.split()[0]  # take the first mirror
+            baseurl = baseurl.rstrip()
+            if not baseurl.endswith('/'):
+                baseurl = baseurl + '/'
```

Comment:
Makes sense to check for trailing slash unconditionally, as it may be left out in repo config as well.